x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -826,6 +860,36 @@ export class TimeSeriesExplorer extends React.Component {
     if (detectorId !== selectedDetectorIndex) {
       appStateHandler(APP_STATE_ACTION.SET_DETECTOR_INDEX, detectorId);
     }
+    // if the detector's function is metric, fetch the highest scoring anomaly record
+    // and set to plot the function_description (avg/min/max) of that record by default
+    if (selectedJob?.analysis_config?.detectors[detectorIndex]?.function === 'metric') {
+      const entityControls = this.getControlsForDetector();
+
+      mlResultsService
+        .getRecordsForCriteria(
+          [selectedJob.job_id],
+          this.getCriteriaFields(selectedDetectorIndex, entityControls),
+          0,
+          null,
+          null,
+          1
+        )
+        .toPromise()
+        .then((resp) => {
+          if (Array.isArray(resp?.records) && resp.records.length === 1) {
+            const highestScoringAnomaly = resp.records[0];
+            this.setState({ actualPlotFunction: highestScoringAnomaly.function_description });
+          }
+        })
+        .catch((resp) => {
+          console.log(
+            'Time series explorer - error getting record with highest anomaly score:',
+            resp
+          );
```

Comment:
instead of logging, we should probably show this error to the user with a toast notification

Reply:
Updated to show toast notification here https://github.com/elastic/kibana/pull/81662/commits/19e20ee5f7a61327b729440a84a8fdf6e87a6a4a